lib/azure/armrest/storage_account_service.rb

```diff
@@ -95,41 +97,27 @@ def list_all
       #   sas = Azure::Armrest::StorageAccountService(config)
       #
       #   sas.create(
-      #     :name     => "yourstorageaccount1",
-      #     :location => "West US",
-      #     :type     => "Standard_ZRS",
-      #     :tags     => {:YourCompany => true}
+      #     "yourstorageaccount1",
+      #     :location   => "West US",
+      #     :tags       => {:YourCompany => true}
+      #     :properties => {:accountType => "Standard_ZRS"},
       #   )
       #
       # For convenience you may also specify the :resource_group as an option.
       #
-      def create(options = {}, rgroup = armrest_configuration.resource_group)
-        rgroup ||= options[:resource_group]
+      def create(name, options = {}, rgroup = armrest_configuration.resource_group)
+        rgroup = options.delete(:resource_group) || rgroup
         raise ArgumentError, "No resource group specified" if rgroup.nil?
 
-        # Mandatory options
-        name = options.fetch(:name)
-        location = options.fetch(:location)
-
-        # Optional
-        tags = options[:tags]
-        type = options[:type] || "Standard_GRS"
+        validating = options.delete(:validating)
 
-        properties = {:accountType => type}
-
-        validate_account_type(type)
+        validate_account_type(options[:properties][:accountType])
         validate_account_name(name)
 
         url = build_url(rgroup, name)
-        url << "&validating=" << options[:validating] if options[:validating]
-
-        body = {
-          :name       => name,
-          :location   => location,
-          :tags       => tags,
-          :properties => properties
-        }.to_json
+        url << "&validating=" << validating if validating
 
+        body = options.to_json
         response = rest_put(url, body)
         response.return!
       end
```

Comment:
Can you take a look at `list_account_keys` and `regenerate_storage_account_keys` methods in this class? `list_account_keys` seems having a bug as `group` is undefined.
The return `regenerate_storage_account_keys` should not use `#return!`. This method is used when we provide a block to the post method to handle response, but not after the call.
I have generalized the `create` method. If the response is an empty string, then return `nil`, otherwise, return the wrapped model object.